mkosi/config.py

```diff
@@ -2909,6 +2900,9 @@ def load_config(args: MkosiArgs, config: argparse.Namespace) -> MkosiConfig:
     if config.output is None:
         config.output = config.image_id or config.image or "image"
 
+        if config.image_version:
+            config.output += f"_{config.image_version}"
```

Comment:
We should move this logic into a `default_factory` for the `output` setting. So add `config_default_output()`, put this logic in there and then use it as the `default_factory` field for the `output` setting. with `default_factory_depends` on the `image_id` and `image_version` settings.